### PR TITLE
civl: re-enable for Linux

### PR DESCRIPTION
**Pull request by iMichka**
Just remove the examples and the docs: in general we do not ship these in our formulae.
Users can still download them from upstream.

- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Review by carlocab — COMMENTED** [review]
You can remove this from the mismatched binary allowlist too. Thanks!

File: Formula/civl.rb [hunk 1]
```diff
@@ -15,18 +16,23 @@ class Civl < Formula
...
+  resource("testdata") do
+    url "https://vsl.cis.udel.edu/lib/sw/civl/1.20/r5259/release/CIVL-1.20_5259.tgz"
+    sha256 "15bf63b3a92300e8432e95397284e29aaa5897e405db9fc2d56cd086f9e330d3"
+  end
+
   def install
     underscored_version = version.to_s.tr("-", "_")
     libexec.install "lib/civl-#{underscored_version}.jar"
     bin.write_jar_script libexec/"civl-#{underscored_version}.jar", "civl"
-    pkgshare.install "doc", "emacs", "examples", "licenses"
```
Comment (carlocab) [inline Formula/civl.rb:26]: We can keep the doc, btw. That doesn't cause any audit failures. Unless the docs take up a large amount of space?

Comment (iMichka) [reply Formula/civl.rb:26]: It's a pdf, I added it back.

Comment (iMichka) [reply Formula/civl.rb:26]: Note: technically speaking I almost consider a pdf a binary file :D

**Comment by iMichka** [comment]
> You can remove this from the mismatched binary allowlist too. Thanks!

It was already not in there.

File: Formula/civl.rb (hunk 1 above)
Comment (carlocab) [reply Formula/civl.rb:26]: That is correct, but not what we mean by "pre-built binaries" 😄

File: Formula/civl.rb [hunk 2]
```diff
@@ -4,7 +4,8 @@ class Civl < Formula
   url "https://vsl.cis.udel.edu/lib/sw/civl/1.20/r5259/release/CIVL-1.20_5259.tgz"
   version "1.20-5259"
   sha256 "15bf63b3a92300e8432e95397284e29aaa5897e405db9fc2d56cd086f9e330d3"
-  revision 1
+  license all_of: ["GPL-3.0-or-later", "LGPL-3.0-or-later", "BSD-3-Clause"]
+  revision 2
```
Comment (carlocab) [inline Formula/civl.rb:8]: Sorry; should've realised this earlier: let's skip the revision bump.
```suggestion
  revision 1
```
Users don't need to reinstall this just to remove the `examples`.

**Comment by carlocab** [comment]
> > You can remove this from the mismatched binary allowlist too. Thanks!
> 
> 
> 
> It was already not in there.

Oh, I was sure I added this because I think this installs x86 macho binaries on ARM too. Weird.

File: Formula/civl.rb (hunk 2 above)
Comment (iMichka) [reply Formula/civl.rb:8]: Yes but we will continue to ship the binaries, which means our CI will continue to fail?

Comment (carlocab) [reply Formula/civl.rb:8]: We can rebuild the bottles if you want. It won't affect CI even if we don't, though.

Comment (iMichka) [reply Formula/civl.rb]: Ok will probably re-bump here: #89642

**Review by carlocab — DISMISSED** [review]

**Comment by iMichka** [comment]
`* 'revision 1' should be removed`. Hu?

**Comment by carlocab** [comment]
Same issue as #88521

**Comment by iMichka** [comment]
> Same issue as #88521

I can add the revision 2 back, and it will fix this, as a side-effect :D

**Comment by carlocab** [comment]
We chose to ignore the audit failure in #88521, so we should probably do the same here.

**Review by SMillerDev — DISMISSED** [review]

**Comment by iMichka** [comment]
Fine, I removed the revision, this can be merged as-is then.

**Review by carlocab — DISMISSED** [review]

File: Formula/civl.rb [hunk 3]
```diff
@@ -40,7 +46,7 @@ def install
       }
     EOS
     # Test with example suggested in manual.
-    example = pkgshare/"examples/concurrency/locksBad.cvl"
+    example = testpath/"examples/concurrency/locksBad.cvl"
```
Comment (carlocab) [inline Formula/civl.rb:49]: Can we just install this fixture into `pkgshare` so we don't need a resource that's identical to the stable tarball?

Comment (iMichka) [reply Formula/civl.rb:49]: In the install block? Won't the audit complain about mismatched binaries being shipped with the bottle?

Comment (carlocab) [reply Formula/civl.rb:49]: Probably not.
```
❯ file locksBad.cvl
locksBad.cvl: C source, ASCII text
```

Comment (iMichka) [reply Formula/civl.rb]: Done. I misread your comment. Should be good now.

**Review by carlocab — APPROVED** [review]

**Comment by carlocab** [comment]
You need to rebottle this so that it uses the `@@HOMEBREW_JAVA@@` placeholder in the bottle.

**Comment by iMichka** [comment]
Looks like the audit prevents us from doing this :(

**Comment by carlocab** [comment]
It would've worked if you had used `pr-publish` here. I suggest you download the bottles, do `pr-upload`, and open a new PR for the bottle commit. (Or you can push straight to master if you want to pretend to be @BrewTestBot)

**Pull request merged**